### DISCUSSION TITLE
Runners using outside-the-US IP addresses?

### DISCUSSION THREAD

**Discussion by rswerve** [community/community#40650] [Actions]
### Select Topic Area

Bug

### Body

I work on a government website that disallows access from outside the US.  We're seeing intermittent failures in our end-to-end tests in CI, and the apparent cause there exactly matches what we see if we try to access our site from outside the States.  Is it possible that our runner is using a non-US IP address?  And is there a way to see the IP address the runner is using?

**Comment by ossanna16** [comment]
Hi there @rswerve and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by vitalykarasik** [comment]
>And is there a way to see the IP address the runner is using?

Sure - like for any linux, for example 

`curl ifconfig.co`

**Reply by rswerve** [reply]
Thanks, yeah, that's what we wound up doing.  The IP is 172.173.174.100, which half of the geolocator services place in Germany, which is what's causing our trouble (the other half put it in Virginia, where I suspect it actually is).

**Reply by vitalykarasik** [reply]
I see.
There are two solutions
- using self-hosted runners
- using github runners with static IP - https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners
